### DISCUSSION TITLE
Installation and Error Issues with SU2 - Adjoint Shape Optimization

### DISCUSSION THREAD

**Discussion by jaime-galiana** [su2code/SU2#2187] [General]
Hello SU2 community,

I am relatively new to SU2 and have encountered challenges during the installation process, specifically when attempting to set up SU2 (v8.0.0) with adjoint shape optimization on Ubuntu 20.04. Despite following the documentation available on the website and building the code from source successfully, I am consistently facing errors when running the shape optimization tutorials.

Here is a summary of the steps I've taken for installation:

Build from source using: 
`./meson.py build -Denable-autodiff=true -Denable-directdiff=true -Dwith-mpi=enabled -Denable-tests=true`

Install using: 
`./ninja -C build install`

The compilation appears to be successful as it creates all the necessary executable files in the 'bin' folder. However, when running the shape optimization tutorials, I encounter different errors (for example: related to missing folders on the path, even when the folders exist. I have checked that I have exported the path correctly and that doesn't seem to be the issue).

Additionally, when running unit tests, I get an error similar to the following:

```
-------------------------------------------------------------------------------
Geometry constructor
-------------------------------------------------------------------------------
../UnitTests/Common/geometry/CGeometry_test.cpp:33
.................................................. .............................

../UnitTests/Common/geometry/CGeometry_test.cpp:42: FAILED:
CHECK( aux_geometry->GetnPoint() == 125 )
with expansion:
63 == 125

../UnitTests/Common/geometry/CGeometry_test.cpp:44: FAILED:
{Unknown expression after the reported line}
due to a fatal error condition:
SIGSEGV - Segmentation violation signal

../UnitTests/Common/geometry/CGeometry_test.cpp:43: FAILED:
CHECK( aux_geometry->GetnElem() == 64 )
with expansion:
43 == 64

../UnitTests/Common/geometry/CGeometry_test.cpp:44: FAILED:
CHECK( aux_geometry->GetnElemHexa() == 64 )
with expansion:
43 == 64

================================================== =============================
test cases: 3 | 2 passed | 1 failed
assertions: 15 | 11 passed | 4 failed

```

It's worth mentioning that the unit tests related to shape optimization pass successfully.

Furthermore, I face difficulties running test cases in parallel due to the MPI configuration. Is there a recommended way to run test cases in parallel when SU2 is compiled with MPI?

I have successfully executed tutorials for compressible and incompressible flow without issues, but encounter problems when attempting more advanced tasks.

Any insights or guidance on resolving these issues and how to install  would be greatly appreciated. Thank you in advance for your assistance!

**Comment by bigfooted** [comment]
Results can depend  a bit on the compiler/platform, the best way to reproduce the regression tests is by running it inside a docker container:
https://su2code.github.io/docs_v7/Container-Development/
